Update word-search extension

j3lte opened the pull request:
## Description

- Updated dependencies to latest versions
- Rewrote API using `useFetch` hook
- Added option to use selection as search text (#11748) (#12045) 
- Added option to set the max results
- Added search for words that sound like the search text
- Added Word Suggestion command, including vocabulary switch (`English`, `Spanish` and `Wikipedia (English)`)
- Added Details option for words

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

j3lte commented:
Added a bunch of new features. Done with all the commits, it can be reviewed now...

RishabSwift commented on extensions/word-search/src/components/SearchResults.tsx:
```diff
@@ -0,0 +1,91 @@
+) {
+  const [search, setSearch] = useState<string>("");
+  const [showDetails, setShowDetails] = useState<boolean>(false);
+  const [vocubalary, setVocubalary] = useState<Vocabulary>(Vocabulary.English);
```
minor spelling `vocabulary, setVocabulary`

RishabSwift commented on extensions/word-search/src/word-suggest.tsx:
```diff
@@ -0,0 +1,45 @@
+export default function SuggestWord(launchProps: LaunchProps) {
+  const placeholder = "Suggest a word";
+
+  const [vocubalary, setVocubalary] = useState<Vocabulary>(Vocabulary.English);
```
minor spelling

RishabSwift commented on extensions/word-search/src/components/SearchResults.tsx:
````diff
@@ -0,0 +1,91 @@
+              <List.Item.Detail
+                markdown={
+                  word.defs !== undefined
+                    ? "```\n" + word.defs.map((d) => d.replaceAll("n\t", "")).join("\n\n") + "```"
````
I think having each definition in its own ``` might look better!
![Screenshot 2024-06-02 at 22 18 00@2x](https://github.com/raycast/extensions/assets/15897093/cc8a524e-ee69-47e7-95ff-6134ee9cf1b9)

What do you think?

```suggestion
                    ? word.defs.map((d) => "```\n" + d.replaceAll("n\t", "") + "\n```").join("\n\n")
```
With this, it looks cleaner. Also, I think there might be a better way to present the tags, because right now they take up half the space. Maybe just an added text at the bottom of the definitions? Or maybe there's a different much better way to present it? What do you think?

RishabSwift reviewed (changes requested):
Wow, so many new features! Thanks for contributing!

Just left a suggestion with the way definitions are presented and the presentation of tags. What do you think?

j3lte commented:
@RishabSwift thanks for the review :). I have updated the code with your suggestions. I also updated the details with a better representation. Now it shows the word in the title and the tags below it, with the definitions in separate rows.

RishabSwift commented:
Perfect! Just tested it out, love the presentation!

Thanks once again

j3lte commented:
@pernielsentikaer can we merge this PR please? 😄

pernielsentikaer reviewed (approved):
Yes we can 🙂
